dask/bytes/core.py

```diff
@@ -281,64 +223,66 @@ class OpenFile(object):
 
     Parameters
     ----------
-    myopen: function
-        Opens the backend file. Should accept path and mode, as the builtin open
-    path: str
+    fs : FileSystem
+        The file system to use for opening the file
+    path : str
         Location to open
-    compression: str or None
-        Compression to apply
-    mode: str like 'rb'
+    mode : str like 'rb', optional
         Mode of the opened file
-    text: bool
+    compression : str or None, optional
+        Compression to apply
+    text : bool, optional
         Whether to wrap the file to be text-like
-    encoding: if using text
-    errors: if using text
+    encoding : str or None, optional
+        The encoding to use if opened in text mode.
+    errors : str or None, optional
+        How to handle encoding errors if opened in text mode.
     """
-    def __init__(self, myopen, path, compression, mode, text, encoding,
+    def __init__(self, fs, path, mode='rb', compression=None, encoding=None,
                  errors=None):
-        self.myopen = myopen
+        self.fs = fs
         self.path = path
-        self.compression = compression
         self.mode = mode
-        self.text = text
+        self.compression = get_compression(path, compression)
         self.encoding = encoding
-        self.closers = None
-        self.fobjects = None
         self.errors = errors
-        self.f = None
+        self.fobjects = []
+
+    def __reduce__(self):
```

Comment:
Just curious, when do you use reduce instead of set/getstate? It indeed seems pretty clean, but I have never done it this way.

Reply:
I prefer `__reduce__` when the underlying collection is just a data container (mostly no-op constructor) as I find it more readable. This is just a stylistic thing though, no strong opinions. The default `object.__reduce__` is actually what handles `__getstate__`/`__setstate__`, `pickle` doesn't know about those methods at all.